tests/unit/spanner_dbapi/test_connection.py

```diff
@@ -595,244 +518,30 @@ def test_rollback_clears_statements(self, mock_transaction):
-        run_mock = connection.run_statement = mock.Mock()
-        run_mock.return_value = ([row], retried_checkum)
-
-        statement = Statement("SELECT 1", [], {}, checksum)
-        connection._statements.append(statement)
-
-        with mock.patch(
-            "google.cloud.spanner_dbapi.connection._compare_checksums"
-        ) as compare_mock:
-            connection.retry_transaction()
-
-        compare_mock.assert_called_with(checksum, retried_checkum)
-        run_mock.assert_called_with(statement, retried=True)
-
-    def test_retry_transaction_w_checksum_mismatch(self):
-        """
-        Check retrying an aborted transaction
-        with results checksums mismatch.
-        """
-        from google.cloud.spanner_dbapi.exceptions import RetryAborted
-        from google.cloud.spanner_dbapi.checksum import ResultsChecksum
-        from google.cloud.spanner_dbapi.parsed_statement import Statement
-
-        row = ["field1", "field2"]
-        retried_row = ["field3", "field4"]
-        connection = self._make_connection()
-
-        checksum = ResultsChecksum()
-        checksum.consume_result(row)
-        retried_checkum = ResultsChecksum()
-        run_mock = connection.run_statement = mock.Mock()
-        run_mock.return_value = ([retried_row], retried_checkum)
-
-        statement = Statement("SELECT 1", [], {}, checksum)
-        connection._statements.append(statement)
-
-        with self.assertRaises(RetryAborted):
-            connection.retry_transaction()
+        self.assertEqual(len(connection._transaction_helper._single_statements), 0)
 
     @mock.patch("google.cloud.spanner_v1.Client")
     def test_commit_retry_aborted_statements(self, mock_client):
         """Check that retried transaction executing the same statements."""
         from google.api_core.exceptions import Aborted
-        from google.cloud.spanner_dbapi.checksum import ResultsChecksum
         from google.cloud.spanner_dbapi.connection import connect
-        from google.cloud.spanner_dbapi.parsed_statement import Statement
-
-        row = ["field1", "field2"]
 
         connection = connect("test-instance", "test-database")
-
-        cursor = connection.cursor()
-        cursor._checksum = ResultsChecksum()
-        cursor._checksum.consume_result(row)
-
-        statement = Statement("SELECT 1", [], {}, cursor._checksum)
-        connection._statements.append(statement)
         mock_transaction = mock.Mock()
         connection._spanner_transaction_started = True
         connection._transaction = mock_transaction
         mock_transaction.commit.side_effect = [Aborted("Aborted"), None]
-        run_mock = connection.run_statement = mock.Mock()
-        run_mock.return_value = ([row], ResultsChecksum())
+        run_mock = connection._transaction_helper = mock.Mock()
 
         connection.commit()
 
-        run_mock.assert_called_with(statement, retried=True)
-
-    @mock.patch("google.cloud.spanner_v1.Client")
```

Comment:
Are all these tests moved to a different file? It makes it very hard to compare what changed and what remained the same. Could you give a TLDR of that?

Reply:
Yes they have been moved to test_transaction_helper.py file along with the source code refactored. I have added more unit tests in test_transaction_helper.py mainly for batch statements which were not present earlier